components/brave_ads/browser/ads_service_impl.h

```diff
@@ -444,6 +446,10 @@ class AdsServiceImpl : public AdsService,
 
   bool is_upgrading_from_pre_brave_ads_build_;
 
+  // This is needed to check if current ads service init become stale as
+  // another ads service start is in progress
+  uint32_t total_number_of_starts_ = 0;
```

Comment:
I think I'm ok with this change if we consider it as a temporary hack for collecting dumps. We'd better craft something more reliable for a longer term (i.e. understand and fix race conditions that cause this discrepancy).

Reply:
@aseren could you please raise a follow up issue assigned to the Ads Project with a label of `ads/feature`

Reply:
@aseren could you please paste a link here for the issue you raised, thanks

Reply:
https://github.com/brave/brave-browser/issues/17643

Reply:
> @aseren based on comments above how about we do not fix the crash but instead collect the dumps if the crash would have occurred?

I have added crash dumps collecting in case if race condition remains. Basing on this we will be able to refactor start/shutdown properly.